public/application.js

```diff
@@ -80,7 +81,69 @@ $(document).ready(function(){
         window.location.href = "/";  
         }
       })
+    } else {
+      window.location.href = "/home";  
     }  
-  })
+  });
+
+  $('#update-details').click(function(event) {
+    $.ajax({
+      type: 'PUT',
+      url: '/users',
+      data: {
+        user: {      
+          password: password.val(),
+          username: username.val(),
+          email: email.val()
+        }
+      },
+      success: function(response){
+        console.log('Info changed', response);
+      }
+    })
+  });
+
+//=====================working on
```

Comment:
to avoid this, you can use [git add](https://git-scm.com/docs/git-add)  --patch for adding parts of a file (`patch` doesn't work on new files tho):

```
git add --patch public/application.js

<<some part of a file appears here with a diff>>

y - stage this hunk
n - do not stage this hunk
q - quit; do not stage this hunk or any of the remaining ones
a - stage this hunk and all later hunks in the file
d - do not stage this hunk or any of the later hunks in the file
g - select a hunk to go to
/ - search for a hunk matching the given regex
j - leave this hunk undecided, see next undecided hunk
J - leave this hunk undecided, see next hunk
k - leave this hunk undecided, see previous undecided hunk
K - leave this hunk undecided, see previous hunk
s - split the current hunk into smaller hunks
e - manually edit the current hunk
? - print help
```

where `hunk` is just a particular block of code that changed at the same time...